clients/java/zpe/src/main/java/com/yahoo/athenz/zpe/AuthZpeClient.java

```diff
@@ -282,7 +289,21 @@ public static void setZPEClientClass(final String className) {
     }
     
     public static PublicKey getZtsPublicKey(String keyId) {
-        return publicKeyStore.getZtsKey(keyId);
+        PublicKey publicKey = publicKeyStore.getZtsKey(keyId);
+        if (publicKey == null && canFetchLatestJwksFromZts()) {
+            //  fetch all zts jwk keys and update config and try again
+            accessSignKeyResolver.loadPublicKeysFromServer();
+            lastZtsJwkFetchTime = System.currentTimeMillis();
+            publicKey = accessSignKeyResolver.getPublicKey(keyId); 
+        }
+        return publicKey;
+    }
+    
+    protected static boolean canFetchLatestJwksFromZts() {
+        int millisBetweenZtsCalls = Integer.parseInt(System.getProperty(ZPE_PROP_MILLIS_BETWEEN_ZTS_CALLS, Integer.toString(30 * 1000 * 60)));
```

Comment:
every time we come across an invalid key we're going to get a system property parse it and then use the value. kind of wasting resources. we should get the system property in the static section for the class and use it all the time.